### PR TITLE
added Skeleton class and updated cards

### PR DESCRIPTION
**Pull request by martyworm**

### CODE REVIEW

File: src/main/java/com/martyworm/cards/individualCards/CardSkeleton.java [hunk 1]
```diff
@@ -27,14 +29,28 @@ public CardSkeleton(Handler handler, int id){
 
     @Override
     public void cast(Tile t){
-        Entity m = handler.getBattle().getEntityManager().addMinion(new RedDragon(handler, 1));
-        m.setXPos(t.getxPos());
-        m.setYPos(t.getyPos());
+
+        if(manaCheck(handler.getBattle().getPlayer1())) {
+
+            Minion m = handler.getBattle().getEntityManager().addMinion(new Skeleton(handler, 2));
```
Comment (EricH-Propser) [inline src/main/java/com/martyworm/cards/individualCards/CardSkeleton.java]: This is a long dot chain, we should talk about this design.

File: src/main/java/com/martyworm/entities/minion/Minion.java [hunk 2]
```diff
@@ -54,8 +54,13 @@ public void tick() {
 
     @Override
     public void render(Graphics g) {
-        g.drawImage(getCurrentAnimationFrame(), xPos-22/*offset the image size to hitBox*/, yPos-22/*offset the image size to hitBox */, width, height, null);//should be width and height not ints
 
+        if(id != 9) {
+            g.drawImage(getCurrentAnimationFrame(), xPos - 13/*offset the image size to hitBox*/, yPos - 10/*offset the image size to hitBox */, width+4, height+4, null);//should be width and height not ints
```
Comment (EricH-Propser) [inline src/main/java/com/martyworm/entities/minion/Minion.java]: you could add a return and remove the following if around the drawimage

File: src/main/java/com/martyworm/Battle/Battle.java [hunk 3]
```diff
@@ -61,21 +64,22 @@ public Battle(Handler handler){
...
                 System.out.println("clicked end turn");
-
+                passTurn();
+                System.out.println("player 1's turn? " + player1.isTurn() + " .. player 2's turn? " + player2.isTurn());
             }
         }));
 
 
         //Will be refactored eventually
 //        entityManager.addEntity(new RedDragon(handler, 1));
```
Comment (EricH-Propser) [inline src/main/java/com/martyworm/Battle/Battle.java]: please don't leave in commented out code.

File: src/main/java/com/martyworm/Battle/Battle.java [hunk 4]
```diff
@@ -145,6 +149,17 @@ public Tile getTile(int tileId){
 
     }
 
+    private void passTurn(){
+        if(player1.isTurn()){
+            player1.setTurn(false);
```
Comment (EricH-Propser) [inline src/main/java/com/martyworm/Battle/Battle.java]: couldn't you just set player1.setTurn(!player1.getTurn()) and same for player 2?

File: src/main/java/com/martyworm/Battle/Battle.java (hunk 3 above)
Comment (martyworm) [reply src/main/java/com/martyworm/Battle/Battle.java]: Removed

File: src/main/java/com/martyworm/Battle/Battle.java (hunk 4 above)
Comment (martyworm) [reply src/main/java/com/martyworm/Battle/Battle.java]: Yea, that's better

File: src/main/java/com/martyworm/cards/individualCards/CardSkeleton.java [hunk 5]
```diff
@@ -5,7 +5,9 @@
 import com.martyworm.board.Tile;
```
[the remark this answers is not in the document]
Comment (martyworm) [reply src/main/java/com/martyworm/cards/individualCards/CardSkeleton.java]: Yea, I've no clue how it's done conventionally.  But each card may and may not have unique variables/traits.  I'm all ears

File: src/main/java/com/martyworm/cards/individualCards/CardSkeleton.java (hunk 1 above)
Comment (martyworm) [reply src/main/java/com/martyworm/cards/individualCards/CardSkeleton.java]: Felt that as I was writing it

File: src/main/java/com/martyworm/entities/minion/Minion.java (hunk 2 above)
Comment (martyworm) [reply src/main/java/com/martyworm/entities/minion/Minion.java]: This should be better once I'm not using these "placeholder" assets.  The spritesheets all have subtle differences in size/location in their respective boxes that are cropped

File: src/main/java/com/martyworm/Battle/Battle.java (hunk 3 above)
Comment (EricH-Propser) [reply src/main/java/com/martyworm/Battle/Battle.java]: You need to push the branch before i can see that.

File: src/main/java/com/martyworm/cards/individualCards/CardSkeleton.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (EricH-Propser) [reply src/main/java/com/martyworm/cards/individualCards/CardSkeleton.java]: Well, designing the data structures you're going to interact with is a valuable exercise. It's always better to think before you code. What are the common things about cards and what are the things that are different? We can probably store that data in JSON, XML or a custom format.

[the remark this answers is not in the document]
Comment (martyworm) [reply src/main/java/com/martyworm/cards/individualCards/CardSkeleton.java]: So the minion cards will all have a mana cost, and be able to summon a minion with movements, spells, health, and the ability to attack.

Then from there they will probably have unique abilities.  I have the CardRedDragon and CardSkeleton extending the MinionCard class. 

Point taken nonetheless.

**Pull request merged**